bag-ui/src/main/java/com/maxshop/viewModel/BagViewModel.kt

```diff
@@ -0,0 +1,127 @@
+package com.maxshop.viewModel
+
+import androidx.annotation.StringRes
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.example.bag_ui.R
+import com.maxshop.mapper.BagProductMapper
+import com.maxshop.mapper.BagProductViewStateMapper
+import com.maxshop.model.RecyclerItem
+import com.maxshop.usecase.DeleteProductFromBagUseCase
+import com.maxshop.usecase.GetBagProductsUseCase
+import com.maxshop.usecase.UpdateBagProductUseCase
+import com.maxshop.util.formatPrice
+import com.maxshop.utils.SingleLiveEvent
+import com.maxshop.viewState.BagProductViewState
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import kotlin.NullPointerException
+
+internal class BagViewModel @Inject constructor(
+    private val getBagProductsUseCase: GetBagProductsUseCase,
+    private val bagProductMapper: BagProductMapper,
+    private val deleteProductFromBagUseCase: DeleteProductFromBagUseCase,
+    private val updateBagProductUseCase: UpdateBagProductUseCase,
+    private val bagProductViewStateMapper: BagProductViewStateMapper
+) : BaseViewModel() {
+
+    sealed class Event {
+        data class OpenProduct(val id: Int) : Event()
+        data class ShowToast(@StringRes val idString: Int) : Event()
+    }
+
+    private val _progressBar = MutableLiveData<Boolean>()
+    val progressBar: LiveData<Boolean> get() = _progressBar
+
+    private val _productsList = MutableLiveData<List<RecyclerItem>>()
+    val productsList: LiveData<List<RecyclerItem>> get() = _productsList
+
+    private val _totalPrice = MediatorLiveData<String>()
+    val totalPrice: LiveData<String> get() = _totalPrice
+
+    private val _event = SingleLiveEvent<Event>()
+    val event: LiveData<Event> get() = _event
+
+    fun getProducts() {
+        viewModelScope.launch {
+            _progressBar.value = true
+            getBagProductsUseCase.execute().map {
+                bagProductMapper.toViewState(it).also {
+                    collectEvent(it)
+                }
+            }.map {
+                bagProductViewStateMapper.toRecyclerItem(it)
+            }.also { _productsList.value = it }
+            _totalPrice.value = calculatePrice().formatPrice()
+            _progressBar.value = false
+        }
+    }
+
+    private suspend fun collectEvent(viewState: BagProductViewState) {
+        viewModelScope.launch {
+            viewState.uiEvent.collect {
+                handleViewStateEvent(it)
+            }
+        }
+
+        _totalPrice.addSource(viewState.amount) {
+            _totalPrice.value = calculatePrice().formatPrice()
```

Comment:
Please incapsulate formatting inside the calculateprice() method. Maybe the method should be renamed then